### PR TITLE
Clear genesis enrollments properly

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
The commit message explains the reasoning.

### CODE REVIEW

File: source/agora/consensus/ValidatorSet.d [hunk 1]
```diff
@@ -657,6 +658,49 @@ unittest
     assert(set.getValidators(enrolls));
     assert(enrolls.length == 1);
     assert(enrolls[0].utxo_key == utxo_hash4);
+
+    // add enrollment at the genesis block:
+    // validates blocks [1 .. 1008] inclusively
+    set.clearExpiredValidators(99999);  // this clears everything
```
Comment (Geod24) [inline source/agora/consensus/ValidatorSet.d]:
```suggestion
    set.clearExpiredValidators(ulong.max);  // this clears everything
```

Very-very nit-picky: Using `ulong.max` might make things self-obvious.

File: source/agora/consensus/ValidatorSet.d [hunk 2]
```diff
@@ -295,9 +295,10 @@ public class ValidatorSet
 
     public void clearExpiredValidators (ulong block_height) @safe nothrow
     {
-        // the smallest enrolled height would be 1, so the passed block height
-        // should be greater than the validator cycle for deleting validators.
-        if (block_height > Enrollment.ValidatorCycle)
+        // the smallest enrolled height would be 0 (genesis block),
+        // so the passed block height should be at minimum the
+        // size of the validator cycle
+        if (block_height >= Enrollment.ValidatorCycle)
```
Comment (Geod24) [inline source/agora/consensus/ValidatorSet.d]: Can we go with an early return here ? It's a bit bigger refactoring but it makes things clearer.

**Review by Geod24 — COMMENTED** [review]
Typo in the commit message: `aA validator`. Also using `#1` created a link to the issue. Otherwise 👍

File: source/agora/consensus/ValidatorSet.d (hunk 2 above)
Comment (AndrejMitrovic) [reply source/agora/consensus/ValidatorSet.d]: Sure, I thought about it too.

**Comment by AndrejMitrovic** [comment]
Thanks, fixed.

**Comment by AndrejMitrovic** [comment]
Apparently `ulong.max` has uncovered an overflow bug in our usage of SQLite.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**